middle_end/flambda2/simplify/simplify_apply_cont_expr.ml

```diff
@@ -103,39 +95,40 @@ let rebuild_apply_cont apply_cont ~args ~rewrite_id uacc ~after_rebuild =
     in
     after_rebuild expr uacc
   in
-  match UE.find_continuation uenv cont with
-  | Linearly_used_and_inlinable
-      { params; handler; free_names_of_handler; cost_metrics_of_handler } ->
-    (* We must not fail to inline here, since we've already decided that the
-       relevant [Let_cont] is no longer needed.
+  let apply_cont_to_expr apply_cont =
+    let cont = AC.continuation apply_cont in
+    match UE.find_continuation uenv cont with
```

Comment:
I think this match deserves at least a small comment explaining that it operates on the continuation that has been shortcutted to (if any), and explaining why it's reasonable/safe to use the number of occurrences of that continuation to do the `Linearly_used_and_inlinable` transformation, even though the shortcut process can change the number of occurrences. Most ntoably, if the continuation is shortcut, the destination of a shortcut will never be `Linearly_used_and_inlinable` (or else it would have been inlined at its callsite, and thus not become a shortcut).